sdk/python/tests/integration/online_store/test_universal_online.py

```diff
@@ -167,11 +170,85 @@ def test_write_to_online_store(environment, universal_data_sources):
     assert df["conv_rate"].iloc[0] == 0.85
 
 
+def _get_online_features_dict_remotely(
+    endpoint: str,
+    features: Union[List[str], FeatureService],
+    entity_rows: List[Dict[str, Any]],
+    full_feature_names: bool = False,
+) -> Dict[str, List[Any]]:
+    """Sends the online feature request to a remote feature server (through endpoint) and returns the feature dict.
+
+    The output should be identical to:
+
+    >>> fs.get_online_features(features=features, entity_rows=entity_rows, full_feature_names=full_feature_names).to_dict()
+
+    This makes it easy to test the remote feature server by comparing the output to the local method.
+
+    """
+    request = {
+        # Convert list of dicts (entity_rows) into dict of lists (entities) for json request
+        "entities": {key: [row[key] for row in entity_rows] for key in entity_rows[0]},
+        "full_feature_names": full_feature_names,
+    }
+    # Either set features of feature_service depending on the parameter
+    if isinstance(features, list):
+        request["features"] = features
+    else:
+        request["feature_service"] = features.name
+    for _ in range(5):
+        # Send the request to the remote feature server and get the response in JSON format
+        response = requests.post(f"{endpoint}/get-online-features", json=request).json()
+        # Retry if the response is internal server error, which can happen when lambda is being restarted
+        if response.get("message") != "Internal Server Error":
+            break
+    else:
+        raise Exception("Failed to get online features from remote feature server")
+
+    # Get rid of unnecessary structure in the response, leaving list of dicts
+    response = [row["fields"] for row in response["field_values"]]
+    # Convert list of dicts (response) into dict of lists which is the format of the return value
+    return {key: [row[key] for row in response] for key in response[0]}
+
+
+def get_online_features_dict(
+    fs: FeatureStore,
+    features: Union[List[str], FeatureService],
+    entity_rows: List[Dict[str, Any]],
+    full_feature_names: bool = False,
+) -> Dict[str, List[Any]]:
+    """Get the online feature values from both SDK and remote feature servers, assert equality and return values.
+
+    Always use this method instead of fs.get_online_features(...) in this test file.
+
+    """
+    online_features = fs.get_online_features(
+        features=features,
+        entity_rows=entity_rows,
+        full_feature_names=full_feature_names,
+    )
+    assertpy.assert_that(online_features).is_not_none()
+    dict1 = online_features.to_dict()
+
+    endpoint = fs.get_feature_server_endpoint()
```

Comment:
should we fail the test if the endpoint doesn't exist and the repo config has `feature_server=True`?